<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Christopher Owen Hernandez" />
  <meta name="author" content="Jack Fabrizio" />
  <meta name="author" content="Joanne Labate" />
  <meta name="author" content="Zhangjun Fei" />
  <meta name="author" content="Michael Mazourek" />
  <meta name="dcterms.date" content="2021-08-26" />
  <meta name="keywords" content="GRIN, Cucurbits, Diversity, Germplasm collection, Cucurbita pepo, Cucurbita moschata, Cucurbita maxima" />
  <title>Characterization of the USDA Cucurbita pepo, Cucurbita moschata, and Cucurbita maxima Collections</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Characterization of the USDA Cucurbita pepo, Cucurbita moschata, and Cucurbita maxima Collections" />
  <meta name="citation_title" content="Characterization of the USDA Cucurbita pepo, Cucurbita moschata, and Cucurbita maxima Collections" />
  <meta property="og:title" content="Characterization of the USDA Cucurbita pepo, Cucurbita moschata, and Cucurbita maxima Collections" />
  <meta property="twitter:title" content="Characterization of the USDA Cucurbita pepo, Cucurbita moschata, and Cucurbita maxima Collections" />
  <meta name="dc.date" content="2021-08-26" />
  <meta name="citation_publication_date" content="2021-08-26" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Christopher Owen Hernandez" />
  <meta name="citation_author_institution" content="Department of Plant Breeding and Genetics, Cornell University, Ithaca, NY" />
  <meta name="citation_author_orcid" content="0000-0002-1668-7121" />
  <meta name="twitter:creator" content="@None" />
  <meta name="citation_author" content="Jack Fabrizio" />
  <meta name="citation_author_institution" content="Department of Plant Breeding and Genetics, Cornell University, Ithaca, NY" />
  <meta name="citation_author_orcid" content="None" />
  <meta name="twitter:creator" content="@None" />
  <meta name="citation_author" content="Joanne Labate" />
  <meta name="citation_author_institution" content="USDA, Geneva, NY" />
  <meta name="citation_author_orcid" content="None" />
  <meta name="twitter:creator" content="@None" />
  <meta name="citation_author" content="Zhangjun Fei" />
  <meta name="citation_author_institution" content="Boyce Thompson Institute for Plant Research, Ithaca, NY" />
  <meta name="citation_author_orcid" content="0000-0001-9684-1450" />
  <meta name="twitter:creator" content="@None" />
  <meta name="citation_author" content="Michael Mazourek" />
  <meta name="citation_author_institution" content="Department of Plant Breeding and Genetics, Cornell University, Ithaca,NY" />
  <meta name="citation_author_orcid" content="0000-0002-2285-7692" />
  <meta name="twitter:creator" content="@None" />
  <link rel="canonical" href="https://ch728.github.io/cucurbit-usda/" />
  <meta property="og:url" content="https://ch728.github.io/cucurbit-usda/" />
  <meta property="twitter:url" content="https://ch728.github.io/cucurbit-usda/" />
  <meta name="citation_fulltext_html_url" content="https://ch728.github.io/cucurbit-usda/" />
  <meta name="citation_pdf_url" content="https://ch728.github.io/cucurbit-usda/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://ch728.github.io/cucurbit-usda/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://ch728.github.io/cucurbit-usda/v/b17d55bac4808669f286152b7e9e47e02b03d3a5/" />
  <meta name="manubot_html_url_versioned" content="https://ch728.github.io/cucurbit-usda/v/b17d55bac4808669f286152b7e9e47e02b03d3a5/" />
  <meta name="manubot_pdf_url_versioned" content="https://ch728.github.io/cucurbit-usda/v/b17d55bac4808669f286152b7e9e47e02b03d3a5/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Characterization of the USDA Cucurbita pepo, Cucurbita moschata, and Cucurbita maxima Collections</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://ch728.github.io/cucurbit-usda/v/b17d55bac4808669f286152b7e9e47e02b03d3a5/">permalink</a>)
was automatically generated
from <a href="https://github.com/ch728/cucurbit-usda/tree/b17d55bac4808669f286152b7e9e47e02b03d3a5">ch728/cucurbit-usda@b17d55b</a>
on August 26, 2021.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Christopher Owen Hernandez</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1668-7121">0000-0002-1668-7121</a><br>
<small>
Department of Plant Breeding and Genetics, Cornell University, Ithaca, NY
</small></p></li>
<li><p><strong>Jack Fabrizio</strong><br><br>
<small>
Department of Plant Breeding and Genetics, Cornell University, Ithaca, NY
</small></p></li>
<li><p><strong>Joanne Labate</strong><br><br>
<small>
USDA, Geneva, NY
</small></p></li>
<li><p><strong>Zhangjun Fei</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9684-1450">0000-0001-9684-1450</a><br>
<small>
Boyce Thompson Institute for Plant Research, Ithaca, NY
</small></p></li>
<li><p><strong>Michael Mazourek</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2285-7692">0000-0002-2285-7692</a><br>
<small>
Department of Plant Breeding and Genetics, Cornell University, Ithaca,NY
</small></p></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>The <em>Cucurbita</em> genus is home to a number of economically and culturally important species. We present the analysis of genotyping-by-sequencing data generated from sequencing the USDA germplasm collections of <em>Cucurbita pepo</em>, <em>Cucurbita moschata</em>, and <em>Cucurbita maxima</em>. These collections include a mixture of wild, landrace, and cultivated specimens from all over the world. Roughly 4,000 - 40,000 quality SNPs were called in each of the collections, which ranged in size from 314 to 829 accessions. Genomic analyses were conducted to characterize the diversity in each of the species and revealed extensive structure corresponding to a combination of geographical origin and morphotype/market class. GWAS was conducted for each data set using both historical and contemporary data, and signals were detected for several traits, including the bush gene (<em>Bu</em>) in <em>C. pepo</em>. These data represent the largest collection of sequence <em>Cucurbita</em> and can be used to direct the maintenance of genetic diversity, develop breeding resources, and to help prioritize whole-genome re-sequencing for further GWAS and other genomics studies aimed at understanding the phenotypic and genetic diversity present in <em>Cucurbita</em>.</p>
<h2 id="introduction">Introduction</h2>
<p>The <em>Cucurbitaceae</em> (Cucurbit) family is home to a number of vining species
mostly cultivated for their fruits. This diverse and economically important
family includes cucumber (<em>Cucumis sativa</em>), melon (<em>Cucumis melo</em>),
watermelon (<em>Citrullus lanatus</em>), and squash (<em>Cucurbita ssp.</em>)
<span class="citation" data-cites="tgxTBPr">[<a href="#ref-tgxTBPr" role="doc-biblioref">1</a>]</span>. Like other cucurbits, squash exhibit
diversity in growth habit, fruit morphology, metabolite content, disease
resistance, and have a nuanced domestication story
<span class="citation" data-cites="17l4Xfem9 AFISuI1C">[<a href="#ref-17l4Xfem9" role="doc-biblioref">2</a>,<a href="#ref-AFISuI1C" role="doc-biblioref">3</a>]</span>.
The genomes of <em>Cucurbita ssp.</em> are
small (roughly 500 Mb), but result from complex interactions between ancient
genomes brought together through an allopolyploidization event
<span class="citation" data-cites="42uqqDAI">[<a href="#ref-42uqqDAI" role="doc-biblioref">4</a>]</span>. These factors make squash an excellent model for
understanding the biology of genomes, fruit development, and domestication.
Within <em>Cucurbita</em>, five species are recognized as domesticated.
Three of these are broadlycultivated: <em>Cucurbita maxima</em>, <em>Cucurbita moschata</em>,
and <em>Cucurbita pepo</em> <span class="citation" data-cites="tgxTBPr">[<a href="#ref-tgxTBPr" role="doc-biblioref">1</a>]</span>.
Few genomic resources have been
available for working with these species; although, draft genomes and
annotations, along with web-based tools and other genomics data are
emerging <span class="citation" data-cites="9Tl87cAk">[<a href="#ref-9Tl87cAk" role="doc-biblioref">5</a>]</span>. Already, these resources have been used to
elucidate the genetics of fruit quality, growth habit, disease resistance, and
to increase the efficiency of cucurbit improvement
<span class="citation" data-cites="7ixiomhx 1FGV3LKSq NJ1zzfWj cccKQqlH 15AR06I5T AmPX0wxj">[<a href="#ref-7ixiomhx" role="doc-biblioref">6</a>,<a href="#ref-1FGV3LKSq" role="doc-biblioref">7</a>,<a href="#ref-NJ1zzfWj" role="doc-biblioref">8</a>,<a href="#ref-cccKQqlH" role="doc-biblioref">9</a>,<a href="#ref-15AR06I5T" role="doc-biblioref">10</a>,<a href="#ref-AmPX0wxj" role="doc-biblioref">11</a>]</span>; however, there has yet to be a
comprehensive survey of the genetic diversity in large diverse <em>Cucurbita</em>
germplasm panels, such as those maintained by the USDA within the Germplasm
Resources Information Network (GRIN) system.</p>
<p>Germplasm collections play a vital role in maintaining and preserving
genetic variation. These collections can be mined by breeders for valuable
alleles and can also be used by geneticists and biologists
for mapping studies <span class="citation" data-cites="we6SIqfY">[<a href="#ref-we6SIqfY" role="doc-biblioref">12</a>]</span>. Like many other orphan and
specialty crops,there has been little effort put into developing community genetic
resources for squash and other cucurbits.
The Cucurbit Coordinated Agricultural Project (CucCap project) was established
to help close the knowledge gap in Cucurbits. This collaborative project aims to
provide genomics resources and tools that can aid in both applied breeding and basic
research. The genetic and phenotypic diversity present in the USDA watermelon and cucumber
collections has already been explored as part of the CucCap project, partially
through the sequencing of USDA germplasm collections and development of core collections for
whole-genome sequencing <span class="citation" data-cites="HsdSI8Y0 BvKcH3LX">[<a href="#ref-HsdSI8Y0" role="doc-biblioref">13</a>,<a href="#ref-BvKcH3LX" role="doc-biblioref">14</a>]</span>.
The diverse specimens of the USDA squash collections have yet to be well characterized at the genetic level;
although, an elaborate system has been established for classifying squash based
on species and various other characteristics.</p>
<p>The classification system used in squash is complex. Squash from each species can be
classed as winter or summer squash depending on whether the fruit is consumed at an
immature or mature stage, the latter is a winter squash <span class="citation" data-cites="vq8Sw1f">[<a href="#ref-vq8Sw1f" role="doc-biblioref">15</a>]</span>.
Squash are considered ornamental if they are used for decoration, and some irregularly shaped,
inedible ornamental squash are called gourds; however, gourds include members of <em>Cucurbita</em>
as well as some species from <em>Lagenaria</em>—not all gourds are squash
<span class="citation" data-cites="UEfO5avO">[<a href="#ref-UEfO5avO" role="doc-biblioref">16</a>]</span>. Many squash are known as pumpkins;
the pumpkin designation is aculture dependent colloquialism that can refer to jack O’ lantern types,
squash used for desserts or, in some Latin American countries, to eating squash from <em>C. moschata</em>
known locally as Calabaza <span class="citation" data-cites="tgxTBPr">[<a href="#ref-tgxTBPr" role="doc-biblioref">1</a>]</span>. Cultivars deemed as pumpkins can be
found in all widely cultivated squash species. Unlike the previous groupings, morophotypes/market
classes are defined within species.For example, a Zucchini is reliably a member of <em>C. pepo</em> and a
Buttercups are from <em>C. maxima</em>. Adding to the complexity of their classification, the <em>Cucurbita</em>
species are believed to have arisen from independent domestication events and the
relationships between cultivated and wild species remains poorly understood <span class="citation" data-cites="Ftv0kDE1">[<a href="#ref-Ftv0kDE1" role="doc-biblioref">17</a>]</span>.</p>
<p><em>C. pepo</em> is the most economically important of the <em>Cucurbita</em> species
and is split into two different subspecies: <em>C. pepo</em> subsp. <em>pepo</em>
and <em>C. pepo</em> subsp. <em>ovifera</em> <span class="citation" data-cites="15AR06I5T">[<a href="#ref-15AR06I5T" role="doc-biblioref">10</a>]</span>. Evidence points
to Mexico as the center of origin for <em>pepo</em> and southwest/central United States
as the origin of <em>ovifera</em>. The progenitor of <em>ovifera</em> is considered by some
to be subsp. <em>ovifera</em> var. <em>texana</em>, whereas subsp. <em>fraterna</em>
is a candidate progenitor for <em>pepo</em> <span class="citation" data-cites="Ftv0kDE1">[<a href="#ref-Ftv0kDE1" role="doc-biblioref">17</a>]</span>. Europe played a
crucial role as a secondary center of diversification for <em>pepo</em>, but not
<em>ovifera</em> <span class="citation" data-cites="50VKpbIa">[<a href="#ref-50VKpbIa" role="doc-biblioref">18</a>]</span>. Important morphoptypes of <em>pepo</em>
include Zucchini, Spaghetti squash, Cocozelle, Vegetable marrow, and some
ornamental pumpkins. <em>C. pepo</em> subsp. <em>ovifera</em> includes summer
squash from the Crookneck, Scallop, and Straightneck group, and winter squash
such as Delicata and Acorn <span class="citation" data-cites="c4MdYrhq">[<a href="#ref-c4MdYrhq" role="doc-biblioref">19</a>]</span>.</p>
<p>The origin of <em>C. moschata</em> is more uncertain than <em>C. pepo</em>;
it is unclear whether <em>C. moschata</em> has a South or North American origin
<span class="citation" data-cites="AFISuI1C">[<a href="#ref-AFISuI1C" role="doc-biblioref">3</a>]</span>. Where and when domestication occurred for this species is
also unknown; however it is known that <em>C. moschata</em> had an India-Myanmar
secondary center of origin where the species was further diversified <span class="citation" data-cites="42uqqDAI">[<a href="#ref-42uqqDAI" role="doc-biblioref">4</a>]</span>.
<em>C. moschata</em> plays an important role in squash breeding as it cross-fertile to
various degrees with <em>C. pepo</em> and <em>C. maxima</em>, and can thus be used as a bridge
to move genes across species <span class="citation" data-cites="42uqqDAI">[<a href="#ref-42uqqDAI" role="doc-biblioref">4</a>]</span>. Popular market classes
of <em>C. moschata</em> include Cheese types like Dickenson, which is widely used
for canned pumpkin products, Butternut (neck) types, Japonica, and tropical
pumpkins known as Calabaza <span class="citation" data-cites="tgxTBPr">[<a href="#ref-tgxTBPr" role="doc-biblioref">1</a>]</span>.</p>
<p><em>C. maxima</em> contains many popular winter squash including Buttercup/Kobocha
types, Kuri, Hubbard, and Banana squash <span class="citation" data-cites="doi">[<a href="#ref-tgxTBPr" role="doc-biblioref">1</a>]</span>. This
species also sports the world’s largest fruit, the giant pumpkin whose fruit are grown for
competition and can reach well over 1000 Kg <span class="citation" data-cites="1lHSz5v1">[<a href="#ref-1lHSz5v1" role="doc-biblioref">20</a>]</span>. Although this
species exhibits a wide range of phenotypic diversity in terms of fruit characteristics,
it appears to be the least genetically diverse of the three species described
<span class="citation" data-cites="Ftv0kDE1">[<a href="#ref-Ftv0kDE1" role="doc-biblioref">17</a>]</span>. <em>C. maxima</em> is believed to have a South American origin,
and was likely domesticated near Peru, with a secondary center of
domestication in Japan/China [nee_domestication_1990; <span class="citation" data-cites="42uqqDAI">[<a href="#ref-42uqqDAI" role="doc-biblioref">4</a>]</span>].</p>
<p>In this study, we set out to characterize the genetic diversity present in
the USDA <em>Cucurbita</em> germplasm collections for <em>C. pepo</em>, <em>C. moschata</em>,
and <em>C. maxima</em>. We present genotyping-by-sequencing data from each of these
collections, population genomics analysis, results from genome-wide association
using historical and contemporary phenotypes, and develop a core panel for re-sequencing.</p>
<h2 id="material-and-methods">Material and Methods</h2>
<h3 id="plant-materials-and-genotyping">Plant Materials and Genotyping</h3>
<p>All available germplasm were requested from USDA cooperators for <em>C. maxima</em> (534), <em>C. moschata</em> (314), and <em>C. pepo</em> (829) respectively. Seeds were planted in 50-cell trays and two 3/4 inch punches of tissue (approximately 80-150 mg) was sampled from the first true leaf of each seedling. DNA was extracted using Omega Mag-Bind Plant DNA DS kits (M1130, Omega Bio-Tek, Norcross, GA) and quantified using Quant-iT PicoGreen dsDNA Kit (Invitrogen, Carlsbad, CA). Purified DNA was shipped to Cornell’s Genomic Diversity Facility for GBS library preparation using protocols optimized for each species. Libraries were sequenced at either 96, 192, or 384-plex on the HiSeq 2500 (Illumina Inc., USA) with single-end mode and a read length of 101 bp.</p>
<h3 id="variant-calling-and-filtering">Variant Calling and Filtering</h3>
<p>SNP calling was conducted using the TASSEL-GBS V5 pipeline <span class="citation" data-cites="LzyXzJq4">[<a href="#ref-LzyXzJq4" role="doc-biblioref">21</a>]</span>. Tags produced by this pipeline were aligned using the default settings of the BWA aligner <span class="citation" data-cites="roi5bwXL">[<a href="#ref-roi5bwXL" role="doc-biblioref">22</a>]</span>. Raw variants were filtered using VCFtools <span class="citation" data-cites="PzHxvkzH">[<a href="#ref-PzHxvkzH" role="doc-biblioref">23</a>]</span>. Before filtering SNPs, samples with a total read depth of <span class="math inline">\(\geq 2\)</span> standard deviations below the mean of all samples were removed before further analysis. Settings for filtering SNPs were as follows, minor allele frequency (MAF) <span class="math inline">\(\geq 0.01\)</span>, missingness <span class="math inline">\(\leq 0.5\)</span>, and biallelic. Three outlier genotypes were found in an initial PCA analysis of the <em>C. maxima</em> data and were removed, as they were likely not <em>C. maxima</em>. Variants were further filtered for specific uses as described below.</p>
<h3 id="population-genomics-analysis">Population Genomics Analysis</h3>
<p>ADMIXTURE <span class="citation" data-cites="9MxZ3Enw">[<a href="#ref-9MxZ3Enw" role="doc-biblioref">24</a>]</span>, which uses a model-based approach to infer ancestral populations (<span class="math inline">\(k\)</span>) and admixture proportions in a given sample, was used to explore population structure in each dataset. ADMIXTURE does not model linkage disequilibrium; thus, marker sets were further filtered to obtain SNPs in approximate linkage equilibrium using the “–indep-pairwise” option in PLINK <span class="citation" data-cites="juGWJJrG">[<a href="#ref-juGWJJrG" role="doc-biblioref">25</a>]</span> with <span class="math inline">\(r^2\)</span> set to 0.1, a window size of 50 SNPs, and a 10 SNP step size . All samples labeled as cultivars were removed from the data prior to running ADMIXTURE. Cross-validation was used to determine the best <span class="math inline">\(k\)</span> value for each species. Briefly, ADMIXTURE was run with different <span class="math inline">\(k\)</span> values (1-20) and the cross-validation error was reported for each <span class="math inline">\(k\)</span>. The <span class="math inline">\(k\)</span> value with minimal cross-validation error was chosen for each species (Supplemental Figures). Ancestral populations were then assigned to cultivars using the program’s projection feature.</p>
<p>Principal components analysis (PCA) was used as a model-free way of determining population structure. The original filtered marker data, not the LD-pruned data used for ADMIXTURE, were converted to a dosage matrix using VCFtool’s “–012” argument. A kinship matrix <span class="math inline">\(\mathbf{K}\)</span> was created using the dosage matrix as input to the “A.mat()” function in Sommer <span class="citation" data-cites="I3vP8cpz">[<a href="#ref-I3vP8cpz" role="doc-biblioref">26</a>]</span>. PCA was conducted using the R function “princomp()” with <span class="math inline">\(\mathbf{K}\)</span> supplied as the covariance matrix.</p>
<p>Phylogenetic analysis was conducted in a subset of the <em>C. pepo</em> panel with clearly labeled subspecies information or where enough information to unambiguously assign the accession to a subspecies was present. The SNPhylo <span class="citation" data-cites="A9obAHi0">[<a href="#ref-A9obAHi0" role="doc-biblioref">27</a>]</span> pipeline was used to infer an unrooted tree using the maximum likelihood method. Default settings were used, except the minimum coverage parameter was decreased to 3 instead of 5 to account for the lower average coverage of GBS data.</p>
<h3 id="analysis-of-phenotypic-data">Analysis of Phenotypic Data</h3>
<p>Historical data were obtained from the USDA Germplasm Resources Information Network (GRIN; http://www.ars-grin.gov) for <em>C. maxima</em>, <em>C. pepo</em>, and <em>C. moschata</em>. All duplicated entries were removed for qualitative traits, where categories are mutually exclusive, leaving only samples with unique entries for analysis. Contemporary phenotypic data were collected from a subset of the <em>C. pepo</em> collection grown in the summer of 2018 in Ithaca, NY. Field-grown plants were phenotyped for vining bush habit at three different stages during the growing seasons to confirm bush, semi-bush or vining growth habit. Plants that had a bush habit early in the season but started to vine at the end of the season were considered semi-bush.</p>
<p>Genomic heritability <span class="citation" data-cites="obxjDYob">[<a href="#ref-obxjDYob" role="doc-biblioref">28</a>]</span> (<span class="math inline">\(h_{g}^2\)</span>) was calculated for all phenotypes. The parameter <span class="math inline">\(h_{g}^2\)</span> was calculated for continuous traits using the formula <span class="math inline">\(h_{g}^2 = \frac{\sigma_{g}^2}{\sigma_{g}^2 + \sigma_{e}^2}\)</span>, where <span class="math inline">\(\sigma_{g}^2\)</span> and <span class="math inline">\(\sigma_{e}^2\)</span> are genetic and error variances estimated from a whole-genome regression of phenotype on marker data using ASReml-R . Multi-class categorical traits were converted to one or several different binary traits depending on the number of entries in each category. For binary traits, a Logit model was fit for the binary response and the heritability was estimated as <span class="math inline">\(h_{g}^2 = \frac{\sigma_{g}^2}{\sigma_{g}^2 + \frac{\pi^2}{3}}\)</span> <span class="citation" data-cites="15ChEFKjS">[<a href="#ref-15ChEFKjS" role="doc-biblioref">29</a>]</span>. In addition to heritability, the amount of phenotypic variance explained by population structure (<span class="math inline">\(R_{pop}^2\)</span>) was calculated from a multiple linear regression of phenotype on sturcture inferred by ADMIXTURE. The R function lm was used to regress continuous phenotypes on the <span class="math inline">\(\mathbf{Q}\)</span> matrix obtained from ADMIXTURE. The R glm function was used with “family=binomial” to regress binary traits on population structure. As there is no <span class="math inline">\(R^2\)</span> defined for logistic models, McFadden’s psuedo <span class="math inline">\(R^2\)</span> was used to assess the correlation between binary traits and population structure <span class="citation" data-cites="1B5fUw44w">[<a href="#ref-1B5fUw44w" role="doc-biblioref">30</a>]</span>.</p>
<h3 id="gwas">GWAS</h3>
<p>Data were imputed prior to association analysis. LinkImpute <span class="citation" data-cites="bh0NyElc">[<a href="#ref-bh0NyElc" role="doc-biblioref">31</a>]</span>, as implemented by the TASSEL <span class="citation" data-cites="6BHN02He">[<a href="#ref-6BHN02He" role="doc-biblioref">32</a>]</span> “LDKNNiImputatioHetV2Plugin” plugin was used for imputation with default settings. Any data still missing after this process were mean imputed. The GENESIS <span class="citation" data-cites="doi">[<a href="#ref-doi" role="doc-biblioref"><strong>doi?</strong></a> 10.1093/bioinformatics/btz567]</span> R package, which can model both binary and continuous traits, was used for association. All models included the first two PCs of the marker matrix as fixed effects and modeled genotype effect (<span class="math inline">\(u\)</span>) as a random effect distributed according to the kinship (<span class="math inline">\(\mathbf{K}\)</span>) matrix (<span class="math inline">\(u \sim N(0, \sigma_{u}^2\mathbf{K})\)</span>). Binary traits were modeled using the logistic regression feature in GENESIS.</p>
<h3 id="syntenty-of-bu-putative-region-in-c.-pepo-and-c.-maxima">Syntenty of <em>Bu</em> putative region in <em>C. pepo</em> and <em>C. maxima</em></h3>
<h3 id="creation-of-a-core-collection">Creation of a Core Collection</h3>
<p>Subsets representative of each panel’s genetic diversity were identified through running GenoCore <span class="citation" data-cites="1BZpw3PUi">[<a href="#ref-1BZpw3PUi" role="doc-biblioref">33</a>]</span> on each of the filtered SNP sets. A subset of the <em>C. pepo</em> panel and key genotypes from the other two species were combined to form a core collection for the cucurbit community. Key genotypes were chosen to represent important market classes and for variation based on variation in traits. These genotypes will be further purified through two additional rounds of selfing and then resequenced using skim-sequecing to produce whole-genome data.</p>
<h2 id="results">Results</h2>
<h3 id="genotyping">Genotyping</h3>
<p>Each <em>Cucurbita ssp.</em> collection was genotyped using the Cornell Genotype by Sequencing (GBS) protocol. This resulted in 534 accessions for <em>C. maxima</em>, 314 for <em>C. moschata</em>, and 829 for <em>C. pepo</em>. Figure <a href="#fig:world">1</a> shows the regional distribution of accessions broken down by species. <em>C. maxima</em> and <em>C. moschata</em> constitute the majority of accessions collected from Central and South America, whereas <em>C. pepo</em> accessions are more prevalent in North America and Europe. <em>C. pepo</em> had the highest number of raw SNPs (108,279) followed by <em>C. moschata</em> (85,345) and <em>C. maxima</em> (56,598). After filtering, <em>C. pepo</em> and <em>C. moschata</em> had a similar number of SNPs, around 40,000, whereas <em>C. maxima</em> had an order of magnitude fewer filtered SNPs (4787). This discrepancy may be an artifact of using Pst1, a rarer base-cutter previously optimized for use in <em>C. maxima</em> <span class="citation" data-cites="hjX723Qb">[<a href="#ref-hjX723Qb" role="doc-biblioref">34</a> ]</span>, rather than ApeK1 which was used for <em>C.pepo</em> and <em>C. moschata</em>. The number and distribution of SNPs across each chromosomes is shown in Table <a href="#tbl:snp_info">1</a>.</p>
<div id="fig:world" class="fignos">
<figure>
<img src="images/pi_map.png" alt="Figure 1: Geographical distribution of the USDA Cucurbita ssp. collection. The size of the pie chart is scaled according to the number of accessions and sector areas correspond to the proportion of the three species." /><figcaption aria-hidden="true"><span>Figure 1:</span> Geographical distribution of the USDA Cucurbita ssp. collection. The size of the pie chart is scaled according to the number of accessions and sector areas correspond to the proportion of the three species.</figcaption>
</figure>
</div>
<div id="tbl:snp_info" class="tablenos">
<table id="tbl:snp_info">
<caption><span>Table 1:</span> Distribution and number of raw and filetered SNPs per chromosome for each species </caption>
<thead>
<tr class="header">
<th>Chrom.</th>
<th><em>C. pepo</em></th>
<th></th>
<th><em>C. moschata</em></th>
<th></th>
<th><em>C. maxima</em></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td>Raw</td>
<td>Filtered</td>
<td>Raw</td>
<td>Filtered</td>
<td>Raw</td>
<td>Filtered</td>
</tr>
<tr class="even">
<td>0</td>
<td>16901</td>
<td>5656</td>
<td>3748</td>
<td>1236</td>
<td>1501</td>
<td>419</td>
</tr>
<tr class="odd">
<td>1</td>
<td>9245</td>
<td>4155</td>
<td>4575</td>
<td>2627</td>
<td>4185</td>
<td>300</td>
</tr>
<tr class="even">
<td>2</td>
<td>6160</td>
<td>2921</td>
<td>4092</td>
<td>2535</td>
<td>2101</td>
<td>169</td>
</tr>
<tr class="odd">
<td>3</td>
<td>5908</td>
<td>2668</td>
<td>3815</td>
<td>2393</td>
<td>2201</td>
<td>157</td>
</tr>
<tr class="even">
<td>4</td>
<td>5540</td>
<td>2652</td>
<td>7868</td>
<td>4458</td>
<td>5703</td>
<td>382</td>
</tr>
<tr class="odd">
<td>5</td>
<td>4813</td>
<td>2254</td>
<td>3226</td>
<td>1804</td>
<td>3115</td>
<td>154</td>
</tr>
<tr class="even">
<td>6</td>
<td>4555</td>
<td>2100</td>
<td>3663</td>
<td>2182</td>
<td>3035</td>
<td>345</td>
</tr>
<tr class="odd">
<td>7</td>
<td>3677</td>
<td>1761</td>
<td>3300</td>
<td>1784</td>
<td>2705</td>
<td>148</td>
</tr>
<tr class="even">
<td>8</td>
<td>4551</td>
<td>2189</td>
<td>2692</td>
<td>1577</td>
<td>2391</td>
<td>191</td>
</tr>
<tr class="odd">
<td>9</td>
<td>4521</td>
<td>1995</td>
<td>3427</td>
<td>1902</td>
<td>2750</td>
<td>229</td>
</tr>
<tr class="even">
<td>10</td>
<td>4366</td>
<td>2052</td>
<td>4219</td>
<td>2225</td>
<td>2297</td>
<td>120</td>
</tr>
<tr class="odd">
<td>11</td>
<td>3839</td>
<td>1727</td>
<td>5212</td>
<td>2962</td>
<td>3713</td>
<td>309</td>
</tr>
<tr class="even">
<td>12</td>
<td>3777</td>
<td>1614</td>
<td>5329</td>
<td>2286</td>
<td>2026</td>
<td>162</td>
</tr>
<tr class="odd">
<td>13</td>
<td>4002</td>
<td>1879</td>
<td>3888</td>
<td>2013</td>
<td>2131</td>
<td>257</td>
</tr>
<tr class="even">
<td>14</td>
<td>4275</td>
<td>1973</td>
<td>5568</td>
<td>3198</td>
<td>4317</td>
<td>297</td>
</tr>
<tr class="odd">
<td>15</td>
<td>3086</td>
<td>1427</td>
<td>3911</td>
<td>2358</td>
<td>2662</td>
<td>172</td>
</tr>
<tr class="even">
<td>16</td>
<td>4274</td>
<td>1589</td>
<td>3407</td>
<td>1987</td>
<td>2058</td>
<td>302</td>
</tr>
<tr class="odd">
<td>17</td>
<td>3519</td>
<td>1657</td>
<td>3557</td>
<td>1888</td>
<td>2195</td>
<td>251</td>
</tr>
<tr class="even">
<td>18</td>
<td>3568</td>
<td>1723</td>
<td>3775</td>
<td>2105</td>
<td>1826</td>
<td>133</td>
</tr>
<tr class="odd">
<td>19</td>
<td>4015</td>
<td>1860</td>
<td>3278</td>
<td>1716</td>
<td>1793</td>
<td>169</td>
</tr>
<tr class="even">
<td>20</td>
<td>3687</td>
<td>1692</td>
<td>3795</td>
<td>1623</td>
<td>1893</td>
<td>133</td>
</tr>
<tr class="odd">
<td>Total</td>
<td>108279</td>
<td>47544</td>
<td>85345</td>
<td>46859</td>
<td>56598</td>
<td>4799</td>
</tr>
</tbody>
</table>
</div>
<h3 id="population-structure-and-genetic-diversity">Population Structure and Genetic Diversity</h3>
<div id="fig:struc" class="fignos">
<figure>
<img src="images/pop_structure.png" alt="Figure 2: Population structure results aligned vertically by species. (A) Admixture plots: each stacked barplot represents an accession colored by proportion of inferred ancestral population. Groups based on hierarchical clustering are delimited by vertical bars and labeled with numbers along the bottom. (B) Plots of the first two principle components (PC) of accessions colored by region, variation explained by PCs is labeled on each axis." /><figcaption aria-hidden="true"><span>Figure 2:</span> Population structure results aligned vertically by species. (A) Admixture plots: each stacked barplot represents an accession colored by proportion of inferred ancestral population. Groups based on hierarchical clustering are delimited by vertical bars and labeled with numbers along the bottom. (B) Plots of the first two principle components (PC) of accessions colored by region, variation explained by PCs is labeled on each axis.</figcaption>
</figure>
</div>
<div id="fig:cult" class="fignos">
<figure>
<img src="images/cult_str.png" alt="Figure 3: Ancestry coefficients projected on cultivars from each species. Results are shown grouped by market/varietal class." /><figcaption aria-hidden="true"><span>Figure 3:</span> Ancestry coefficients projected on cultivars from each species. Results are shown grouped by market/varietal class.</figcaption>
</figure>
</div>
<p>Filtered SNPs were used for population structure analysis. Available geographical, phenotypic, and other metadata were retrieved from GRIN and were used to help interpret structure results. Results from model-based admixture analysis are shown in Figure <a href="#fig:struc">2</a> panel A. These data support five ancestral groups (K=5) in each of the species. Population structure was driven mostly by geography, except in <em>C. pepo</em> where the presence of different subspecies was responsible for some of the structure. Commonalities among structure groups are described in Table . The first two principal components (PCs) derived from principal components analysis (PCA) of the marker data are shown in Figure <a href="#fig:struc">2</a> panel B. As with the model-based analysis, PCA showed geography as a main driver of population structure with accessions being derived from Africa, the Arab States, Asia, Europe, North America, and South/Latin America. PC1 in <em>C. pepo</em> separates <em>C. pepo</em> subsp.<em>ovifera</em>, which have a North American Origin, from subsp. <em>pepo</em>.</p>
<div id="fig:phylo" class="fignos">
<figure>
<img src="images/pepo_phylo.png" alt="Figure 4: Unrooted maximum likelihood tree of C. pepo subspecies inferred using wild and cultivated germplasm in the C. pepo collection." /><figcaption aria-hidden="true"><span>Figure 4:</span> Unrooted maximum likelihood tree of <em>C. pepo</em> subspecies inferred using wild and cultivated germplasm in the <em>C. pepo</em> collection.</figcaption>
</figure>
</div>
<p>Ancestry proportions from admixture analysis were projected onto cultivars/market types identified in the accessions, which were excluded from the initial analysis used to infer ancestral groups. Cultivars were grouped according to known market class within species to help identify patterns in ancestry among and between market classes. Key market types identified in accessions from <em>C. pepo</em> including Acorn, Scallop, Crook, Pumpkin (jacko’ lantern), Zuchinni, Marrow, Gem, and Spaghetti; Neck, Cheese, Japonica, and Calabaza in <em>C. moschata</em>; and Buttercup, Kobocha, Kuri, Hubbard, and Mammoth (show squash) in <em>C. maxima</em>. These groupings are shown in Figure <a href="#fig:cult">3</a>. In general, members of each market class exhibit similar ancestry proportions. In <em>C. pepo</em> market classes from the two different subspecies had distinct ancestry patterns. For example, Acorn, Scallop and Crook market classes are all from subsp. <em>ovifera</em> and all of these classes had similar ancestry proportions with roughly 50% of ancestry from the wild <em>ovifera</em>. In contrast, market classes within <em>pepo</em> had a small percentage of ancestry from wild <em>ovifera</em> and more ancestry in common with European and Asian accessions. With <em>C. moschata</em>, Neck and Cheese type market classes showed very similar ancestry patterns, whereas the Japonica and Calabaza types were more distinct. Relative to the <em>C. pepo</em> and <em>C. moschata</em>, the <em>C. maxima</em> cultivars were less distinct from one another.</p>
<p>Unlike <em>C. moschata</em> and <em>C. maxima</em>, several different subspecies were present in the <em>C. pepo</em> collection, including some wild specimens. A group of 82 accessions with 11,065 high-quality SNPs was used for constructing an unrooted phylogenetic tree. The tree is shown in Figure <a href="#fig:phylo">4</a> and recapitulates the relationships among <em>pepo</em> subspecies shown in previous work <span class="citation" data-cites="Ftv0kDE1">[<a href="#ref-Ftv0kDE1" role="doc-biblioref">17</a>]</span>.</p>
<h3 id="analysis-of-phenotypic-data-1">Analysis of Phenotypic Data</h3>
<p>All available historical data from GRIN were compiled. Only traits with <span class="math inline">\(\geq\)</span> 100 entries were considered for further analysis. Filtering resulted in 21 traits for <em>C. pepo</em>, 5 for <em>C. moscahta</em> and 16 for <em>C. maxima</em>. Traits spanned fruit and agronomic-related characteristics, as well as pest resistances. The number of records for a given trait ranged from 108 to 822, with an average of <span class="math inline">\(\sim\)</span> 270. Fruit traits included fruit width, length, surface color and texture, and flesh color and thickness. Agronomic data included plant vigor and vining habit, and several phenotypes related to maturity. Pest-related traits included susceptibility to cucumber beetle and squash bug in <em>C. pepo</em> and watermelon mosaic virus (WMV) and powdery mildew (PM) in <em>C. maxima</em>.</p>
<p>of traits measured on a quantitative scale were normally distributed. Marker-based narrow-sense heritability (<span class="math inline">\(h_{G}^2\)</span>) was calculated for each trait. Values for <span class="math inline">\(h_{G}^2\)</span> ranged from 0.12 to close to 1. Most traits had moderate to high heritabilities (<span class="math inline">\(\geq\)</span> 0.4). Regression of trait data on the <span class="math inline">\(\mathbf{Q}\)</span> matrix obtained from structure analysis was used to determine the amount of phenotypic variation explained by population structure. In <em>C. pepo</em>, traits related to fruit morphology tended to have high correlations with population structure (<span class="math inline">\(R_{pop}^2\)</span>). Seed weight had the highest correlation with an <span class="math inline">\(R_{pop}^2\)</span> of 0.6. In <em>C. moschata</em>, maturity showed the highest correlation with population structure (<span class="math inline">\(R_{pop}^2\)</span> of 0.52). None of the 16 traits in <em>C. maxima</em> had a high correlation with population structure. The only exception was plant growth habit.Traits related to pest resistance were measure in <em>C. maxima</em> and <em>C. pepo</em> and had among the lowest correlations with population structure.</p>
<h3 id="genome-wide-association">Genome-wide Association</h3>
<div id="fig:gwas" class="fignos">
<figure>
<img src="images/bush_gene_assoc.png" alt="Figure 5: GWAS result for the Bush gene (Bu) in C. pepo" /><figcaption aria-hidden="true"><span>Figure 5:</span> GWAS result for the Bush gene (<em>Bu</em>) in <em>C. pepo</em></figcaption>
</figure>
</div>
<p>Genome-wide association was conducted for all traits using standard mixed-model analysis. No significant signals were detected in <em>C. moschata</em>. A weak signal was detected in <em>C. maxima</em> for fruit set on chromosome 12 and fruit ribbing on chromosome 17. Three phenotypes were significantly associated with SNPs in <em>C. pepo</em>: bush/vine plant architecture on chromosome 10, fruit flesh color on chromosome 5, and fruit width on chromosome 3. The bush/vine phenotype exhibited the strongest signal, and the Manhatten plot and p-value quantile-quantile plot is shown in Figure <a href="#fig:gwas">5</a>.</p>
<h3 id="syntenty-of-bu-putative-region-in-c.-pepo-and-c.-maxima-1">Syntenty of <em>Bu</em> putative region in <em>C. pepo</em> and <em>C. maxima</em></h3>
<h3 id="development-of-a-core-collection">Development of a Core Collection</h3>
<p>A core set of accessions that covered over 99% of total genetic diversity was identified in each of the panels. Roughly 10 to 20% of the accessions were required to capture the genetic diversity in the panels (See Supplemental Figures ). This amounted to 245 accessions in <em>C. pepo</em>, 154 in <em>C. moschata</em>, and in 248 <em>C.maxima</em>. The core subset identified in <em>C. pepo</em> was augmented with accessions that represented key market classes or that had traits of interest to breeding programs. Additionally, key accessions were selected from <em>C. maxima</em>, <em>C. moschata</em> and some wild species. Together these genotypes were purified through two additional rounds of selfing and seed will serve as the basis for a <em>Cucurbita ssp.</em> core to be used by breeding programs and researchers for further studies.</p>
<h2 id="discussion">Discussion</h2>
<h2 id="supplemental">Supplemental</h2>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-tgxTBPr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Pumpkin and Winter Squash</strong> <div class="csl-block">María Ferriol, Belén Picó</div> <em>Springer Science and Business Media LLC</em> (2007-12-06) <a href="https://doi.org/dmqkmf">https://doi.org/dmqkmf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-0-387-30443-4_10">10.1007/978-0-387-30443-4_10</a></div></div>
</div>
<div id="ref-17l4Xfem9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>The Genes of Pumpkin and Squash</strong> <div class="csl-block">Harry S Paris, Rebecca Nelson Brown</div> <em>HortScience</em> (2005-10) <a href="https://doi.org/gmkkfh">https://doi.org/gmkkfh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.21273/hortsci.40.6.1620">10.21273/hortsci.40.6.1620</a></div></div>
</div>
<div id="ref-AFISuI1C" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Origin and domestication of Cucurbitaceae crops: insights from phylogenies, genomics and archaeology</strong> <div class="csl-block">Guillaume Chomicki, Hanno Schaefer, Susanne S Renner</div> <em>New Phytologist</em> (2019-08) <a href="https://doi.org/gsg7">https://doi.org/gsg7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/nph.16015">10.1111/nph.16015</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31230355">31230355</a></div></div>
</div>
<div id="ref-42uqqDAI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Karyotype Stability and Unbiased Fractionation in the Paleo-Allotetraploid Cucurbita Genomes</strong> <div class="csl-block">Honghe Sun, Shan Wu, Guoyu Zhang, Chen Jiao, Shaogui Guo, Yi Ren, Jie Zhang, Haiying Zhang, Guoyi Gong, Zhangcai Jia, … Yong Xu</div> <em>Molecular Plant</em> (2017-10) <a href="https://doi.org/gb4cx2">https://doi.org/gb4cx2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.molp.2017.09.003">10.1016/j.molp.2017.09.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28917590">28917590</a></div></div>
</div>
<div id="ref-9Tl87cAk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Cucurbit Genomics Database (CuGenDB): a central portal for comparative and functional genomics of cucurbit crops</strong> <div class="csl-block">Yi Zheng, Shan Wu, Yang Bai, Honghe Sun, Chen Jiao, Shaogui Guo, Kun Zhao, Jose Blanca, Zhonghua Zhang, Sanwen Huang, … Zhangjun Fei</div> <em>Nucleic Acids Research</em> (2019-01-08) <a href="https://doi.org/gmcmq9">https://doi.org/gmcmq9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gky944">10.1093/nar/gky944</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30321383">30321383</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6324010">PMC6324010</a></div></div>
</div>
<div id="ref-7ixiomhx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>An SNP-based saturated genetic map and QTL analysis of fruit-related traits in Zucchini using Genotyping-by-sequencing</strong> <div class="csl-block">Javier Montero-Pau, José Blanca, Cristina Esteras, Eva Ma Martínez-Pérez, Pedro Gómez, Antonio J Monforte, Joaquín Cañizares, Belén Picó</div> <em>BMC Genomics</em> (2017-01-18) <a href="https://doi.org/gmkkvf">https://doi.org/gmkkvf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12864-016-3439-y">10.1186/s12864-016-3439-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28100189">28100189</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5241963">PMC5241963</a></div></div>
</div>
<div id="ref-1FGV3LKSq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>A high-density linkage map and QTL mapping of fruit-related traits in pumpkin (Cucurbita moschata Duch.)</strong> <div class="csl-block">Yu-Juan Zhong, Yang-Yang Zhou, Jun-Xing Li, Ting Yu, Ting-Quan Wu, Jian-Ning Luo, Shao-Bo Luo, He-Xun Huang</div> <em>Scientific Reports</em> (2017-10-06) <a href="https://doi.org/gmkktr">https://doi.org/gmkktr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41598-017-13216-3">10.1038/s41598-017-13216-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28986571">28986571</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5630576">PMC5630576</a></div></div>
</div>
<div id="ref-NJ1zzfWj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Genetic mapping of ovary colour and quantitative trait loci for carotenoid content in the fruit of Cucurbita maxima Duchesne</strong> <div class="csl-block">Karolina Kaźmińska, Ewelina Hallmann, Anna Rusaczonek, Aleksandra Korzeniewska, Mirosław Sobczak, Joanna Filipczak, Karol Seweryn Kuczerski, Jarosław Steciuk, Monika Sitarek-Andrzejczyk, Marek Gajewski, … Grzegorz Bartoszewski</div> <em>Molecular Breeding</em> (2018-08-27) <a href="https://doi.org/gd6tc4">https://doi.org/gd6tc4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s11032-018-0869-z">10.1007/s11032-018-0869-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30237748">30237748</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6133072">PMC6133072</a></div></div>
</div>
<div id="ref-cccKQqlH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Genomic Prediction of Pumpkin Hybrid Performance</strong> <div class="csl-block">Po‐Ya Wu, Chih‐Wei Tung, Chieh‐Ying Lee, Chen‐Tuo Liao</div> <em>The Plant Genome</em> (2019-06) <a href="https://doi.org/gmkkvg">https://doi.org/gmkkvg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3835/plantgenome2018.10.0082">10.3835/plantgenome2018.10.0082</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31290920">31290920</a></div></div>
</div>
<div id="ref-15AR06I5T" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Whole-genome resequencing of Cucurbita pepo morphotypes to discover genomic variants associated with morphology and horticulturally valuable traits</strong> <div class="csl-block">Aliki Xanthopoulou, Javier Montero-Pau, Ifigeneia Mellidou, Christos Kissoudis, José Blanca, Belén Picó, Aphrodite Tsaballa, Eleni Tsaliki, Athanasios Dalakouras, Harry S Paris, … Ioannis Ganopoulos</div> <em>Horticulture Research</em> (2019-08-11) <a href="https://doi.org/gmkkvd">https://doi.org/gmkkvd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41438-019-0176-9">10.1038/s41438-019-0176-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31645952">31645952</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6804688">PMC6804688</a></div></div>
</div>
<div id="ref-AmPX0wxj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Genomic Prediction and Selection for Fruit Traits in Winter Squash</strong> <div class="csl-block">Christopher O Hernandez, Lindsay E Wyatt, Michael R Mazourek</div> <em>G3 Genes|Genomes|Genetics</em> (2020-10-01) <a href="https://doi.org/gmkzhb">https://doi.org/gmkzhb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1534/g3.120.401215">10.1534/g3.120.401215</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32816923">32816923</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7534422">PMC7534422</a></div></div>
</div>
<div id="ref-we6SIqfY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Mobilizing Crop Biodiversity</strong> <div class="csl-block">Susan McCouch, Zahra Katy Navabi, Michael Abberton, Noelle L Anglin, Rosa Lia Barbieri, Michael Baum, Kirstin Bett, Helen Booker, Gerald L Brown, Glenn J Bryan, … Loren H Rieseberg</div> <em>Molecular Plant</em> (2020-10) <a href="https://doi.org/gmkzrd">https://doi.org/gmkzrd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.molp.2020.08.011">10.1016/j.molp.2020.08.011</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32835887">32835887</a></div></div>
</div>
<div id="ref-HsdSI8Y0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>The USDA cucumber (Cucumis sativus L.) collection: genetic diversity, population structure, genome-wide association studies, and core collection development</strong> <div class="csl-block">Xin Wang, Kan Bao, Umesh K Reddy, Yang Bai, Sue A Hammar, Chen Jiao, Todd C Wehner, Axel O Ramírez-Madera, Yiqun Weng, Rebecca Grumet, Zhangjun Fei</div> <em>Horticulture Research</em> (2018-10-01) <a href="https://doi.org/gfdjfd">https://doi.org/gfdjfd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41438-018-0080-8">10.1038/s41438-018-0080-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30302260">30302260</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6165849">PMC6165849</a></div></div>
</div>
<div id="ref-BvKcH3LX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>Genome of ‘Charleston Gray’, the principal American watermelon cultivar, and genetic characterization of 1,365 accessions in the U.S. National Plant Germplasm System watermelon collection</strong> <div class="csl-block">Shan Wu, Xin Wang, Umesh Reddy, Honghe Sun, Kan Bao, Lei Gao, Linyong Mao, Takshay Patel, Carlos Ortiz, Venkata L Abburi, … Zhangjun Fei</div> <em>Plant Biotechnology Journal</em> (2019-05-07) <a href="https://doi.org/gmkktt">https://doi.org/gmkktt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/pbi.13136">10.1111/pbi.13136</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31022325">31022325</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6835170">PMC6835170</a></div></div>
</div>
<div id="ref-vq8Sw1f" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Morpho-Physiological Aspects of Productivity and Quality in Squash and Pumpkins ( &lt;i&gt;Cucurbita&lt;/i&gt; spp.)</strong> <div class="csl-block">JBrent Loy</div> <em>Critical Reviews in Plant Sciences</em> (2004-07) <a href="https://doi.org/abs/10.1080/07352680490490733">https://doi.org/abs/10.1080/07352680490490733</a> <div class="csl-block">DOI: <a href="https://doi.org/abs/10.1080/07352680490490733">abs/10.1080/07352680490490733</a></div></div>
</div>
<div id="ref-UEfO5avO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Germplasm enhancement of Cucurbita pepo (pumpkin, squash, gourd: Cucurbitaceae): progress and challenges</strong> <div class="csl-block">Harry S Paris</div> <em>Euphytica</em> (2015-11-24) <a href="https://doi.org/f8ds6k">https://doi.org/f8ds6k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10681-015-1605-y">10.1007/s10681-015-1605-y</a></div></div>
</div>
<div id="ref-Ftv0kDE1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Evolutionary and domestication history of Cucurbita (pumpkin and squash) species inferred from 44 nuclear loci</strong> <div class="csl-block">Heather R Kates, Pamela S Soltis, Douglas E Soltis</div> <em>Molecular Phylogenetics and Evolution</em> (2017-06) <a href="https://doi.org/f97dq2">https://doi.org/f97dq2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ympev.2017.03.002">10.1016/j.ympev.2017.03.002</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28288944">28288944</a></div></div>
</div>
<div id="ref-50VKpbIa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Italian horticultural and culinary records of summer squash ( &lt;i&gt;Cucurbita pepo&lt;/i&gt; , Cucurbitaceae) and emergence of the zucchini in 19th-century Milan</strong> <div class="csl-block">Teresa A Lust, Harry S Paris</div> <em>Annals of Botany</em> (2016-07) <a href="https://doi.org/gmkk6b">https://doi.org/gmkk6b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/aob/mcw080">10.1093/aob/mcw080</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27343231">27343231</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4934399">PMC4934399</a></div></div>
</div>
<div id="ref-c4MdYrhq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>Parallel Evolution Under Domestication and Phenotypic Differentiation of the Cultivated Subspecies of Cucurbita pepo (Cucurbitaceae)</strong> <div class="csl-block">Harry S Paris, Ales Lebeda, Eva Křistkova, Thomas C Andres, Michael H Nee</div> <em>Economic Botany</em> (2012-01-31) <a href="https://doi.org/fzc57g">https://doi.org/fzc57g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s12231-012-9186-3">10.1007/s12231-012-9186-3</a></div></div>
</div>
<div id="ref-1lHSz5v1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>The making of giant pumpkins: how selective breeding changed the phloem of &lt;i&gt;C&lt;/i&gt; &lt;i&gt;ucurbita maxima&lt;/i&gt; from source to sink</strong> <div class="csl-block">JESSICA A SAVAGE, DUSTIN F HAINES, NMICHELE HOLBROOK</div> <em>Plant, Cell &amp; Environment</em> (2015-08) <a href="https://doi.org/f7jhh7">https://doi.org/f7jhh7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/pce.12502">10.1111/pce.12502</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25546629">25546629</a></div></div>
</div>
<div id="ref-LzyXzJq4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>TASSEL-GBS: A High Capacity Genotyping by Sequencing Analysis Pipeline</strong> <div class="csl-block">Jeffrey C Glaubitz, Terry M Casstevens, Fei Lu, James Harriman, Robert J Elshire, Qi Sun, Edward S Buckler</div> <em>PLoS ONE</em> (2014-02-28) <a href="https://doi.org/f5zjsk">https://doi.org/f5zjsk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0090346">10.1371/journal.pone.0090346</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24587335">24587335</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3938676">PMC3938676</a></div></div>
</div>
<div id="ref-roi5bwXL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>Fast and accurate short read alignment with Burrows-Wheeler transform</strong> <div class="csl-block">H Li, R Durbin</div> <em>Bioinformatics</em> (2009-05-18) <a href="https://doi.org/dqt59j">https://doi.org/dqt59j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btp324">10.1093/bioinformatics/btp324</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19451168">19451168</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2705234">PMC2705234</a></div></div>
</div>
<div id="ref-PzHxvkzH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>The variant call format and VCFtools</strong> <div class="csl-block">P Danecek, A Auton, G Abecasis, CA Albers, E Banks, MA DePristo, RE Handsaker, G Lunter, GT Marth, ST Sherry, … 1000 Genomes Project Analysis Group</div> <em>Bioinformatics</em> (2011-06-07) <a href="https://doi.org/b6kxfd">https://doi.org/b6kxfd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btr330">10.1093/bioinformatics/btr330</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21653522">21653522</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137218">PMC3137218</a></div></div>
</div>
<div id="ref-9MxZ3Enw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>Enhancements to the ADMIXTURE algorithm for individual ancestry estimation</strong> <div class="csl-block">David H Alexander, Kenneth Lange</div> <em>BMC Bioinformatics</em> (2011-06-18) <a href="https://doi.org/dtnztg">https://doi.org/dtnztg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-12-246">10.1186/1471-2105-12-246</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21682921">21682921</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3146885">PMC3146885</a></div></div>
</div>
<div id="ref-juGWJJrG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses</strong> <div class="csl-block">Shaun Purcell, Benjamin Neale, Kathe Todd-Brown, Lori Thomas, Manuel AR Ferreira, David Bender, Julian Maller, Pamela Sklar, Paul IW de Bakker, Mark J Daly, Pak C Sham</div> <em>The American Journal of Human Genetics</em> (2007-09) <a href="https://doi.org/cp2rzn">https://doi.org/cp2rzn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1086/519795">10.1086/519795</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17701901">17701901</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1950838">PMC1950838</a></div></div>
</div>
<div id="ref-I3vP8cpz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>Genome-Assisted Prediction of Quantitative Traits Using the R Package sommer</strong> <div class="csl-block">Giovanny Covarrubias-Pazaran</div> <em>PLOS ONE</em> (2016-06-06) <a href="https://doi.org/ggjp6v">https://doi.org/ggjp6v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0156744">10.1371/journal.pone.0156744</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27271781">27271781</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4894563">PMC4894563</a></div></div>
</div>
<div id="ref-A9obAHi0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>SNPhylo: a pipeline to construct a phylogenetic tree from huge SNP data</strong> <div class="csl-block">Tae-Ho Lee, Hui Guo, Xiyin Wang, Changsoo Kim, Andrew H Paterson</div> <em>BMC Genomics</em> (2014) <a href="https://doi.org/f5xgnj">https://doi.org/f5xgnj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2164-15-162">10.1186/1471-2164-15-162</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24571581">24571581</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3945939">PMC3945939</a></div></div>
</div>
<div id="ref-obxjDYob" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>Genomic Heritability: What Is It?</strong> <div class="csl-block">Gustavo de los Campos, Daniel Sorensen, Daniel Gianola</div> <em>PLOS Genetics</em> (2015-05-05) <a href="https://doi.org/gmkw9r">https://doi.org/gmkw9r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1005048">10.1371/journal.pgen.1005048</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25942577">25942577</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4420472">PMC4420472</a></div></div>
</div>
<div id="ref-15ChEFKjS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Genome-enabled predictions for binomial traits in sugar beet populations</strong> <div class="csl-block">Filippo Biscarini, Piergiorgio Stevanato, Chiara Broccanello, Alessandra Stella, Massimo Saccomani</div> <em>BMC Genetics</em> (2014) <a href="https://doi.org/f6jhz3">https://doi.org/f6jhz3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2156-15-87">10.1186/1471-2156-15-87</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25053450">25053450</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4113669">PMC4113669</a></div></div>
</div>
<div id="ref-1B5fUw44w" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Log-likelihood-based Pseudo- &lt;i&gt;R&lt;/i&gt; &lt;sup&gt;2&lt;/sup&gt; in Logistic Regression</strong> <div class="csl-block">Giselmar AJ Hemmert, Laura M Schons, Jan Wieseke, Heiko Schimmelpfennig</div> <em>Sociological Methods &amp; Research</em> (2016-03-18) <a href="https://doi.org/gdzt5m">https://doi.org/gdzt5m</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1177/0049124116638107">10.1177/0049124116638107</a></div></div>
</div>
<div id="ref-bh0NyElc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>LinkImpute: Fast and Accurate Genotype Imputation for Nonmodel Organisms</strong> <div class="csl-block">Daniel Money, Kyle Gardner, Zoë Migicovsky, Heidi Schwaninger, Gan-Yuan Zhong, Sean Myles</div> <em>G3 Genes|Genomes|Genetics</em> (2015-11-01) <a href="https://doi.org/f79ns2">https://doi.org/f79ns2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1534/g3.115.021667">10.1534/g3.115.021667</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26377960">26377960</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4632058">PMC4632058</a></div></div>
</div>
<div id="ref-6BHN02He" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>TASSEL: software for association mapping of complex traits in diverse samples</strong> <div class="csl-block">PJ Bradbury, Z Zhang, DE Kroon, TM Casstevens, Y Ramdoss, ES Buckler</div> <em>Bioinformatics</em> (2007-06-22) <a href="https://doi.org/fdj9qg">https://doi.org/fdj9qg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btm308">10.1093/bioinformatics/btm308</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17586829">17586829</a></div></div>
</div>
<div id="ref-1BZpw3PUi" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>GenoCore: A simple and fast algorithm for core subset selection from large genotype datasets</strong> <div class="csl-block">Seongmun Jeong, Jae-Yoon Kim, Soon-Chun Jeong, Sung-Taeg Kang, Jung-Kyung Moon, Namshin Kim</div> <em>PLOS ONE</em> (2017-07-20) <a href="https://doi.org/gbn84s">https://doi.org/gbn84s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0181420">10.1371/journal.pone.0181420</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28727806">28727806</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5519076">PMC5519076</a></div></div>
</div>
<div id="ref-hjX723Qb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>A high-density genetic map for anchoring genome sequences and identifying QTLs associated with dwarf vine in pumpkin (Cucurbita maxima Duch.)</strong> <div class="csl-block">Guoyu Zhang, Yi Ren, Honghe Sun, Shaogui Guo, Fan Zhang, Jie Zhang, Haiying Zhang, Zhangcai Jia, Zhangjun Fei, Yong Xu, Haizhen Li</div> <em>BMC Genomics</em> (2015-12-24) <a href="https://doi.org/gb3hqt">https://doi.org/gb3hqt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12864-015-2312-8">10.1186/s12864-015-2312-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26704908">26704908</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4690373">PMC4690373</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen())
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script>
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
