## Results

### Genotyping

Each *Cucurbita ssp.* collection was genotyped using the Cornell Genotype by Sequencing (GBS) protocol. This resulted in 534 accessions for *C. maxima*, 314 for *C. moschata*, and 829 for *C. pepo*. Figure {@fig:world} shows the regional distribution of accessions broken down by species. *C. maxima* and *C. moschata* constitute the majority of accessions collected from Central and South America, whereas *C. pepo* accessions are more prevalent in North America and Europe. *C. pepo* had the highest number of raw SNPs (108,279) followed by *C. moschata* (85,345) and *C. maxima* (56,598). After filtering, *C. pepo* and *C. moschata* had a similar number of SNPs, around 40,000, whereas *C. maxima* had an order of magnitude fewer filtered SNPs (4787). This discrepancy may be an artifact of using Pst1, a rarer base-cutter previously optimized for use in *C. maxima* [@doi:10.1186/s12864-015-2312-8 ], rather than ApeK1 which was used for *C.pepo* and *C. moschata*. The number and distribution of SNPs across each chromosomes is shown in Table {@tbl:snp_info}. 

![Geographical distribution of the USDA Cucurbita ssp. collection. The size of the pie chart is scaled according to the number of accessions and sector areas correspond to the proportion of the three species.](images/pi_map.png){#fig:world} 

|Chrom. | *C. pepo* |  |*C. moschata* | | *C. maxima*| |
|----------|----------|----------|----------|----------|----------|----------|
| |Raw | Filtered| Raw | Filtered| Raw | Filtered |
|0 |16901|5656|3748|1236|1501|419|
|1 |9245|4155|4575|2627|4185|300|
|2 |6160|2921|4092|2535|2101|169|
|3 |5908|2668|3815|2393|2201|157|
|4 |5540|2652|7868|4458|5703|382|
|5 |4813|2254|3226|1804|3115|154|
|6 |4555|2100|3663|2182|3035|345|
|7 |3677|1761|3300|1784|2705|148|
|8 |4551|2189|2692|1577|2391|191|
|9 |4521|1995|3427|1902|2750|229|
|10 |4366|2052|4219|2225|2297|120|
|11 |3839|1727|5212|2962|3713|309|
|12 |3777|1614|5329|2286|2026|162|
|13 |4002|1879|3888|2013|2131|257|
|14 |4275|1973|5568|3198|4317|297|
|15 |3086|1427|3911|2358|2662|172|
|16 |4274|1589|3407|1987|2058|302|
|17 |3519|1657|3557|1888|2195|251|
|18 |3568|1723|3775|2105|1826|133|
|19 |4015|1860|3278|1716|1793|169|
|20 |3687|1692|3795|1623|1893|133|
|Total|108279|47544|85345|46859|56598|4799|

Table: Distribution and number of raw and filetered SNPs per chromosome for each species {#tbl:snp_info}

### Population Structure and Genetic Diversity

![Population structure results aligned vertically by species. (A) Admixture plots: each stacked barplot represents an accession colored by proportion of inferred ancestral population. Groups based on hierarchical clustering are delimited by vertical bars and labeled with numbers along the bottom. (B) Plots of the first two principle components (PC) of accessions colored by region, variation explained by PCs is labeled on each axis.](images/pop_structure.png){#fig:struc}


|Group|Species|||
|----|----|----|----|
||*C. pepo*|*C. moschata*|*C. maxima*|
|1|Europe/Asia, mostly for Turkey|South American/Latin American|Mixed origin; kobocha/turban types|
|2|Europe, mostly from Macedonia|South American/Latin American|European, mostly from Macedonia|
|3|North America, wild and landrace *ovifera*|African|Asia |Pacific/Middle East|
|4|Mixed origin|India|South American|
|5|South America, mostly from Mexico|Mixed origin; elongated fruit type|African|

Table: Commonalities among accessions in each group, most groupings are dictated by geography. {#tbl:struc_info}

![Ancestry coefficients projected on cultivars from each species. Results are shown grouped by market/varietal class.](images/cult_str.png){#fig:cult}


Filtered SNPs were used for population structure analysis. Available geographical, phenotypic, and other metadata were retrieved from GRIN and were used to help interpret structure results. Results from model-based admixture analysis are shown in Figure {@fig:struc} panel A. These data support five ancestral groups (K=5) in each of the species. Population structure was driven mostly by geography, except in *C. pepo* where the presence of different subspecies was responsible for some of the structure. Commonalities among structure groups are described in Table {@tbl:struc_info}. The first two principal components (PCs) derived from principal components analysis (PCA) of the marker data are shown in Figure {@fig:struc} panel B. As with the model-based analysis, PCA  showed geography as a main driver of population structure with accessions being derived from Africa, the Arab States, Asia, Europe, North America, and South/Latin America. PC1 in *C. pepo* separates *C. pepo* subsp.*ovifera*, which have a North American Origin, from subsp. *pepo*. 


![Unrooted maximum likelihood tree of *C. pepo* subspecies inferred using wild and cultivated germplasm in the *C. pepo* collection.](images/pepo_phylo.png){#fig:phylo}

Ancestry proportions from admixture analysis were projected onto cultivars/market types identified in the accessions, which were excluded from the initial analysis used to infer ancestral groups. Cultivars were grouped according to known market class within species to help identify patterns in ancestry among and between market classes. Key market types identified in accessions from *C. pepo* including Acorn, Scallop, Crook, Pumpkin (jacko' lantern), Zuchinni, Marrow, Gem, and Spaghetti; Neck, Cheese, Japonica, and Calabaza in *C. moschata*; and Buttercup, Kobocha, Kuri, Hubbard, and Mammoth (show squash) in *C. maxima*. These groupings are shown in Figure {@fig:cult}. In general, members of each market class exhibit similar ancestry proportions. In *C. pepo* market classes from the two different subspecies had distinct ancestry patterns. For example, Acorn, Scallop and Crook market classes are all from subsp. *ovifera* and all of these classes had similar ancestry proportions with roughly 50% of ancestry from the wild *ovifera*. In contrast, market classes within *pepo* had a small percentage of ancestry from wild *ovifera* and more ancestry in common with European and Asian accessions. With *C. moschata*, Neck and Cheese type market classes showed very similar ancestry patterns, whereas the Japonica and Calabaza types were more distinct. Relative to the *C. pepo* and *C. moschata*, the *C. maxima* cultivars were less distinct from one another.

Unlike *C. moschata* and *C. maxima*, several different subspecies were present in the *C. pepo* collection, including some wild specimens. A group of 82 accessions with 11,065 high-quality SNPs was used for constructing an unrooted phylogenetic tree. The tree is shown in Figure {@fig:phylo} and recapitulates the relationships among *pepo* subspecies shown in previous work [@doi:10.1016/j.ympev.2017.03.002].

### Analysis of Phenotypic Data

All available historical data  from GRIN were compiled. Only traits with $\geq$ 100 entries were considered for further analysis. Filtering resulted in 21 traits for *C. pepo*, 5 for *C. moscahta* and 16 for *C. maxima*. Traits spanned fruit and agronomic-related characteristics, as well as pest resistances. The number of records for a given trait ranged from 108 to 822, with an average of  $\sim$ 270. Fruit traits included fruit width, length, surface color and texture, and flesh color and thickness. Agronomic data included plant vigor and vining habit, and several phenotypes related to maturity. Pest-related traits included susceptibility to cucumber beetle and squash bug in *C. pepo* and watermelon mosaic virus (WMV)  and powdery mildew (PM) in *C. maxima*. 

of traits measured on a quantitative scale were normally distributed. Marker-based narrow-sense heritability ($h_{G}^2$) was calculated for each trait. Values for $h_{G}^2$ ranged from 0.12 to close to 1. Most traits had moderate to high heritabilities ($\geq$ 0.4). Regression of trait data on the $\mathbf{Q}$ matrix obtained from structure analysis was used to determine the amount of phenotypic variation explained by population structure. In *C. pepo*, traits related to fruit morphology tended to have high correlations with population structure ($R_{pop}^2$). Seed weight had the highest correlation with an $R_{pop}^2$ of 0.6. In *C. moschata*, maturity showed the highest correlation with population structure ($R_{pop}^2$ of 0.52). None of the 16 traits in *C. maxima* had a high correlation with population structure. The only exception was plant growth habit.Traits related to pest resistance were measure in *C. maxima* and *C. pepo* and had among the lowest correlations with population structure.

### Genome-wide Association

![GWAS result for the Bush gene (*Bu*) in *C. pepo*](images/bush_gene_assoc.png){#fig:gwas}

Genome-wide association was conducted for all traits using standard mixed-model analysis. No significant signals were detected in *C. moschata*. A weak signal was detected in *C. maxima* for fruit set on chromosome 12 and fruit ribbing on chromosome 17. Three phenotypes were significantly associated with SNPs in *C. pepo*: bush/vine plant architecture on chromosome 10, fruit flesh color on chromosome 5, and fruit width on chromosome 3. The bush/vine phenotype exhibited the strongest signal, and the Manhatten plot and p-value quantile-quantile plot is shown in Figure {@fig:gwas}.

### Syntenty of *Bu* putative region in *C. pepo* and *C. maxima*

### Development of a Core Collection

A core set of accessions that covered over 99\% of total genetic diversity was identified in each of the panels. Roughly 10 to 20\% of the accessions were required to capture the genetic diversity in the panels (See Supplemental Figures \ref{core}). This amounted to 245 accessions in *C. pepo*, 154 in  *C. moschata*, and in 248 *C.maxima*. The core subset identified in *C. pepo* was augmented with accessions that represented key market classes or that had traits of interest to breeding programs. Additionally, key accessions were selected from *C. maxima*, *C. moschata* and some wild species. Together these genotypes were purified through two additional rounds of selfing and seed will serve as the basis for a *Cucurbita ssp.* core to be used by breeding programs and researchers for further studies.
