## Discussion

*Cucurbita pepo*, *Cucurbita moschata*, and *Cucurbita maxima*, 
exhibit a wide range of phenotypic diversity.
This diversity was evident in the GRIN phenotypic records for these species.
We have demonstrated that there is a wide range of genetic diversity through
genotyping-by-sequencing and genetic analysis of available specimens from 
the germplasm collections. Thousands to tens of thousands of whole-genome markers
where obtained for each species. Clustering of samples and admixture analysis
produced results that align closely with known secondary centers of origin in all species. 
This was especially clear in our analysis of the *Cucurbita pepo* collection.
*Cucurbita pepo* has its origin in the new world, with a secondary center of
diversification in Europe. This pattern was conspicuous in the our PCA analysis.  
Phylogenetic anlaysis of *Cucurbita pepo* using the whole-genome
markers also supported the known relationships between the various subspecies in *pepo*. 
Together with the mapping of a putative bush gene (*Bu*) that appears to be syntenic 
with the bush gene mapped in *C. maxima*, we have demonstrated that these data 
constitute a new, high quality genetic resource for the Cucurbit community. 
These markers and our analysis of available germplasm
have a number of uses for breeding and can also be used to guide future experments
aimed at biological insight.

These data could be used to develop germplasm
specific marker panels for use in breeding applications, as has been done in
other crops {@doi:10.1186/s12284-019-0311-0}.
Possible breeding applications would include marker assisted selection, 
marker assisted backcrossing, and purity assessment of seedstock using a low density panel; 
whereas, a medium density panel could be developed for routine genomic selection.  Our clustering
of samples based on maker data suggest geography is a key driver for overall population 
structure. When projecting ancestry proportions onto cultivars of known market classes, the ancestry 
proportions were relatively similar within market class grouping. Alghough there is genetic diversity 
within each species, this diversity is constrained within market classes. This suggests that
crosses between these market classes would greatly increase the amount of genetic diversity 
to be leveraged in breeding efforts. Crossing between market classes would come at 
the cost of bringing in undesirable characteristics with regards to achieving a specific 
morpho-type associated market class. This cost could be mitgated through the use of
markers to recover morpho-type expediciously during pre-breeding. Ultimately, the 
judicious infusion of diversity into a breeding program is necessary for 
sustaining long-term gain.

Genomic selection (GS) was proposed over twenty years ago {@doi:10.1093/genetics/157.4.1819}, 
and has since become a standard breeding technique. 
Yet, to our knowledge, GS is not used to any
appreciable degree by any of the public-sector breeding programs working with 
cucurbits. Studies specifically looking at GS in squash have demonstrated, as with 
every other crop, that GS is a viable breeding method; although
the specific implementation may vary for each program and must take into 
account the nature of the trait being predicted {@doi:doi.org/10.1534/g3.120.401215 ; 
@doi:doi.org/10.3390/agronomy10060847 ; @doi:10.3835/plantgenome2018.10.0082}. Since
cucurbit crops are more seed-limited than space-limited, a predict-part-test-part or sparse
testing strategy is an obvious starting point {@doi.org/10.3389/fpls.2021.658978}. Selective
phenotyping of resource-intensive quality traits based on marker data to enable prediction
is also low-hanging fruit. Our work lowers the barrier to entry for GS in squash, 
as it provides a set of markers that can be filtered idependently by interested breeding programs, 
rapidly convered into an amplicon-based assay, and tested in target germplasm. This set
can then be used for routine genotyping, which is a necessary first step towards implementing
GS {@doi:10.3389/fpls.2020.00353}.

Outside of breeding, these data can be used for association studies. In the case where
traits are common in the panel, the panel can be phenotyped for a trait of interest and
combined with marker data and insight provided by our study. 
We demonstrated this approach in our association analysis of the bush
gene. In the case of a rare phenotype, such as a resistance gene, subsets of the germplasm 
and markers should be used to develop custom populations. Plant 
introductions (PI) are frequently used as source parent in mapping studies and for germplasm 
improvement, as was the case for mapping Phytophthora resistance and developing resistant
breeding lines (@doi:10.1007/s00122-020-03747-1; @doi:10.21273/HORTTECH04636-20). In cases,
were higher density genotyping may be necessary or the PIs are not well charaterized for a
trait of interest, the data generated in this study can be used to prioritize accessions for
re-sequencing and phentyping. Our geno-core analysis, provides a subset of several hundred
accessions that would likely be informative for re-sequencing efforts.

Unlike many other outcrossing monoicous crops such as, maize and onion, cultivars from Cucurbita, 
similar to sunflower, do not suffer from debilatating inbreeding depression resulting in a 
severe reduction in vigor and fertility (Whitaker and Davis, 1962). 
With little inbreeding depression, it would stand that little better-parent heterosis could 
be achieved considering the dominance theory of heterosis. 
There  were initial papers that suggested that inbreeding in Cucurbita may not simply reduce 
yeild as inbred varieties have the capacity to compete with commercial check cultivars;
however, better-parent heterosis has been observed in a Cucurbita pepo and C
ucurbita maxima (Firpo et. al., 1998). 
Although better-parent heterosis has been observed, inbred cultivars remain competitive 
with there hybrid counter-parts. 
It has been suggested that the oddity of Cucurbita not suffering high levels of 
inbreeding depression is the result of an archaic population bottleneck. 
The population bottleneck would effectively lead to greater levels of population 
homogeneity through many variants in the initial population being lost and the 
low-frequency alleles post-bottleneck having a greater probability of becoming lost. 





